examples/gallery/images/grdgradient_shading.py

```diff
@@ -0,0 +1,62 @@
+"""
+Calculating grid gradient with custom ``azimuth`` and ``normalize`` parameters
+------------------------------------------------------------------------------
+The :func:`pygmt.grdgradient` function calculates the gradient of a grid file.
+As input, :func:`pygmt.grdgradient` gets a :class:`xarray.DataArray` object or
+a path string to a grid file. It then calculates the respective gradient and
+returns an :class:`xarray.DataArray` object. The example below sets two main
+parameters:
+
+- ``azimuth``: to set the illumination light source direction (0° is North,
+  90° is East, 180° is South, 270° is West).
+- ``normalize`` to enhance the three-dimensional sense of the topography.
+
+The ``normalize`` parameter calculates the azimuthal gradient of each point
+along a certain azimuth angle, then adjusts the brightness value of the color
+according to the positive/negative of the azimuthal gradient and the amplitude
+of each point.
+"""
+
+import pygmt
+
+# Load the 3 arc-minutes global relief grid in the target area around Caucasus
+grid = pygmt.datasets.load_earth_relief(resolution="03m", region=[35, 50, 35, 45])
+
+fig = pygmt.Figure()
+
+# Define a colormap to be used for topography
+pygmt.makecpt(cmap="terra", series=[-7000, 7000])
+
+# Define figure configuration
+pygmt.config(FONT_TITLE="10p,5", MAP_TITLE_OFFSET="1p", MAP_FRAME_TYPE="plain")
+
+# Setup subplot panels with three rows and four columns
+with fig.subplot(
+    nrows=3,
+    ncols=4,
+    figsize=("28c", "21c"),
+    sharex="b",
+    sharey="l",
+):
+    # E.g. "0/90" illuminates light source from the North (top) and East
+    # (right), and so on
+    for azi in ["0/90", "0/300", "180/225"]:
+        # "e" and "t" are cumulative Laplace distribution and cumulative
+        # Cauchy distribution, respectively
+        # "amp" (e.g. 1 or 10) controls the brightness value of the color
+        for nor in ["t1", "e1", "t10", "e10"]:
+            # Making an intensity DataArray using azimuth and normalize
+            # parameters
+            shade = pygmt.grdgradient(grid=grid, azimuth=azi, normalize=nor)
+            fig.grdimage(
+                grid=grid,
+                shading=shade,
+                projection="M?",
+                frame=["a4f2", f"+tazimuth={azi}, normalize={nor}"],
+                cmap=True,
+                panel=True,
+            )
+
+fig.colorbar(position="x9.0c/-1.1c+w10c/0.25c+h", frame="a2000f500+lElevation (m)")
```

Comment:
I was expecting using `position="JBC+w10c/0.25c+h"` can place the colorbar at the bottom center of the whole figure, but the output is not what I expected:

![map](https://user-images.githubusercontent.com/3974108/225205587-df709fe2-fd19-441a-8a7f-69e9235cf09e.png)

However, a similar GMT CLI script shows that `JBC` actually works as I expected, so there may be something wrong with PyGMT. It's unrelated to this PR, so I will open a separate issue after this PR is merged.

Test script:
```bash
gmt begin map
	gmt set MAP_FRAME_TYPE=plain
	gmt makecpt -Cterra -T-7000/7000
	gmt subplot begin 3x4 -Ff28c/21c -SRl -SCb
	for i in $(seq 1 12); do
		gmt basemap -R35/50/35/45 -JM? -c
	done
	gmt subplot end
	gmt colorbar -DJBC+w10c/0.25c+h -B2000f500+l'Elevation (m)' -Vd
gmt end show
```
Its output:

![map](https://user-images.githubusercontent.com/3974108/225206048-f2523b18-4af2-4a27-ae1a-67bd7c137c8a.png)

Reply:
Can reproduce the issue, but currently I have no idea why and where this happens.

Reply:
Thanks @seismann for pointing out this issue!
So far, I found that it is somehow related to the values of the tuple passed to the `figsize` parameter:

_Code example_:
```python
# orientated on: https://www.pygmt.org/dev/tutorials/advanced/subplots.html
# last access: 2023/03/15
import pygmt

max_rows = 3
max_cols = 4

fig = pygmt.Figure()

with fig.subplot(
        nrows=max_rows,
        ncols=max_cols,
        #figsize=("15c", "6c"),  # works as expected
        figsize=("28c", "21c"),  # does NOT work as expected
        sharex="b",
        sharey="l",
):
    
    for i in range(max_cols):
    
        for j in range(max_rows):
        
            fig.basemap(
                projection="X?",
                region=[0, 1, 0, 1],
                panel=True,                  
            )

pygmt.makecpt(cmap="berlin", series=[0, 100, 10])
fig.colorbar(position="JBC+h")

fig.show()
#fig.savefig(fname="subplot_colorbar_position_15x6.png")
#fig.savefig(fname="subplot_colorbar_position_28x21.png")
```

_Output figure_:
**figsize=("15c", "6c")**
![subplot_colorbar_position_15x6](https://user-images.githubusercontent.com/94163266/225274950-de8d7ea7-eee4-43ac-8168-b51ac125ee16.png)

**figsize=("28c", "21c")**
![subplot_colorbar_position_28x21](https://user-images.githubusercontent.com/94163266/225274990-b3c45e5f-0566-466b-807b-ee9c703fec3f.png)

Reply:
I just opened an issue in https://github.com/GenericMappingTools/pygmt/issues/2426.